### ISSUE TITLE
Pod failing to start with sidecar injector if any label contains slashes in the key name

### ISSUE THREAD

**Issue by cezarsa** [istio/istio#9571]
**Describe the bug**

When starting a new Pod the istio sidecar injector is creating an environment variable for each label in the format `ISTIO_META_<label key>=<label value>` , but if any label contains `/` as part of its key name the pod fails to be created because it lead to an invalid environment variable name.

This problem apparently was introduced in #9208 and is related to #9338.

**Expected behavior**

The pod should be created successfully.

**Steps to reproduce the bug**

On an environment with istio sidecar injection enabled run:
```
cat <<EOF | kubectl  apply -f -
apiVersion: v1
kind: Pod
metadata:
    name: mypod
    labels:
        com.test/foo: bar
spec:
    containers:
      - name: mypod1
        image: busybox
EOF
```

The following error is returned:

> The Pod "mypod" is invalid: spec.containers[1].env[6].name: Invalid value: "ISTIO_META_com.test/foo": a valid environment variable name must consist of alphabetic characters, digits, '_', '-', or '.', and must not start with a digit (e.g. 'my.env-name',  or 'MY_ENV.NAME',  or 'MyEnvName1', regex used for validation is '[-._a-zA-Z][-._a-zA-Z0-9]*')

**Version**

The problem in question is present both on current master and on tag 1.0.3:

https://github.com/istio/istio/blob/1.0.3/install/kubernetes/helm/istio/templates/sidecar-injector-configmap.yaml#L140-L142

https://github.com/istio/istio/blob/a78d9c527b5ddf709ac86011977ffdf38ce08db7/install/kubernetes/helm/istio/templates/sidecar-injector-configmap.yaml#L165-L167

**Installation**

Using helm chart.

**Comment by plwhite** [comment]
It appears that this means that no StatefulSet can be automatically instrumented through sidecar injection because of the label `statefulset.kubernetes.io/pod-name`. There may be a workaround but I haven't found one.

**Comment by plwhite** [comment]
It appears (based on an initial test of istio, so not impossible I have misunderstood) that this prevents automatic sidecar injection into stateful sets, which have an automagically added label `statefulset.kubernetes.io/pod-name`. That may impact priority.

**Comment by shubb30** [comment]
This also prevents Istio from being used with Rancher-created Kubernetes clusters since Rancher creates labels like `workload.user.cattle.io/workloadselector`.

**Comment by mandarjog** [comment]
This has been addressed by #9634 and #9678

**Comment by mandarjog** [comment]
Addressed by #9634 #9678
